### ISSUE TITLE
How to run Part-1: BoxGen Networks.

### ISSUE THREAD

**Issue by mitsuki05** [RyuZhihao123/Plant-Recon-25#6]
Request for Execution Steps for Part-1: BoxGen Networks
I am currently unable to proceed with the execution of "Part-1: BoxGen Networks." Please provide guidance on the following:

1. IFN Module Dataset Creation
I need instructions on how to prepare the dataset for the IFN module. Specifically:

Data Format: What format should the input data be in (images, masks, etc.)?

Creation Process: How should the dataset files be organized and created to ensure successful training?

Ideally, I would appreciate it if you could provide a small, working dataset for immediate testing, which would greatly help in validating the setup.

2. DEC Module () Locationtrain_box.py
The location of the training script for the DEC module, train_box.py, is unclear. Could you please specify where this file is located within the cloned GitHub repository?

**Comment by RyuZhihao123** [comment]
[Example format](https://github.com/RyuZhihao123/Plant-Recon-25/tree/main/BoxStruNN/data/partnetdata/plant_hier) is over here.
**But I would say it's better to run Part-2 first, because the generation of training dataset for Part-1 is **heavily relied on** the Part-2.** Many people start from Part-1 and thus always encounter many issues.

For example, you can first use my [PlantDataGenerator](https://github.com/RyuZhihao123/Modular-Tree-Modeler-25) to generate a few 3D plant models ([demo video is here](https://drive.google.com/file/d/1FugtnOsYm2L_HkJawwiUUKoe6aQs5e93/view?usp=sharing)). Then, load the saved OBJ/FBX files in **Unity3D** and parse it through [the following C# function](https://github.com/RyuZhihao123/Plant-Recon-25/blob/main/BoxPlantModeling/Assets/Scripts/Utils/Geometry.cs#L1032) (please locate to it to see the script code):

 ```c#
static public (Dictionary<MinBoundingBox, (int, bool)>, int) GetMinBBoxFromTreeRoot(Internode plantRoot)
```

This C# function will automatically transfer your input plants into the corresponding training format. And the ``JsonDatasetGenerator.SaveJson2(plant, "PATH")`` will save them under specified path.

Training and inference of box decoders are over here [[1]](https://github.com/RyuZhihao123/Plant-Recon-25/blob/main/BoxStruNN/code/eval_gen_box.py) and [[2]](https://github.com/RyuZhihao123/Plant-Recon-25/blob/main/BoxStruNN/code/train_dec.py).


### Last.
(1) To ease the usage, I am now working on a **Configure-Free version for this project** (it will be something like my **PlantDataGenerator**), as it's obvious that this project involves too many seperate dependencies (sometimes Python, somethimes C#, and even a little bit C++), and they are not organized very well and with too many code lines (I personally think they are spaghetti code, sadly). So it is annoying to configure them by other people. (This summer, I took 3 weeks to help another person from UT Delft to configure on their PC, but it is exhausting). 

Ideally, **with the configure-free version, you would be able to run an EXE software direclty on the Windows PCs without setting up the environments**, as long as the internet is connected for invoking neural nets on my online server.
Thus, at that time you can also use it to run the demo.

Currently, this version is almost finished. But recently I have to write my thesis. I will release it after **Middle Novemeber**. I will notice you again once it's done.

(2) For  **[PlantDataGenerator](https://github.com/RyuZhihao123/Modular-Tree-Modeler-25)**. **It is a free software** that I developed and it can be directly execuated in Windows PCs without any configuration for generating 3D plant data. 
One thing is that because I am making money though this software (3 entities are using it), thus the code is not released.
But it doesn't affect your normal use since the EXE software file has been uploaded to the GitHub for you to use directly. You can normally generate 3D plants and save their 3D models **for free** without any restrictions. 
Only when you want to modify and customize the code for secondary development, then a payment is required.


If you encounter other problems in terms of configuration, it's welcome to leave messages under this issue, or just contact me through my outlook emal if you want.

**Comment by mitsuki05** [comment]
Thank you for the detailed explanation.
I am really looking forward to the final release after Middle November and the Configure-Free version.
Please let us know when it's ready. Thanks again!
